app/execution.js

```diff
@@ -146,21 +165,35 @@ exports.loadDataAndSendMail = function loadDataAndSendMail(reports_data) {
 
             generate_chart.then(function (response) {
                 var imagefilename = reports_data['report_obj']['report_name'] + '.jpg';
-                wkhtmltoimage.generate(response, { output: image_dir + imagefilename });
+
+                var isGenerateImage = true;
+                if (reports_data.report_line_obj.viz_type == "KPI" || reports_data.report_line_obj.viz_type == "Text Object") {
+                    isGenerateImage = false;
```

Comment:
why this flag isGenerateImage is required

Reply:
@johibkhan2 yes isGenerateImage flag is required because we are converting charts response to image and display that image in report mail but if report type is KPI or text object then no need to convert charts response to image that's why we need isGenerateImage  flag to verify it.